src/Raven.Studio/typescript/components/pages/database/settings/documentSchema/DocumentSchema.tsx

```diff
@@ -0,0 +1,423 @@
+                        schemaValidatorCollections={allCollectionNames}
+                        onSubmit={(data) => handleNewSchemaSubmit(id, data)}
+                        onCancel={() => handleCancelNew(id)}
+                    />
+                ))}
+            </>
+        );
+    }
+
+    return (
+        <>
+            <LazyLoad active={asyncLoadValidators.loading}>
+                {filteredValidators.map((v) => (
+                    <CollectionSchemaRichPanel
+                        schemaValidatorCollections={allCollectionNames}
+                        key={v.Name}
+                        collectionName={v.Name}
+                        schema={v.Schema}
+                    />
+                ))}
+            </LazyLoad>
+            {draftIds.map((id) => (
+                <NewCollectionSchemaRichPanel
+                    key={id}
+                    schemaValidatorCollections={allCollectionNames}
+                    onSubmit={(data) => handleNewSchemaSubmit(id, data)}
+                    onCancel={() => handleCancelNew(id)}
+                />
+            ))}
+        </>
+    );
+};
+
+interface CollectionSchemaRichPanelProps {
+    collectionName: string;
+    schema: string;
+    schemaValidatorCollections: string[];
+}
+
+const CollectionSchemaRichPanel = ({
+    collectionName = "",
+    schema = "",
+    schemaValidatorCollections,
+}: CollectionSchemaRichPanelProps) => {
+    const { value: isEditingSchema, toggle: toggleEditingSchema } = useBoolean(false);
+    const databaseName = useAppSelector(databaseSelectors.activeDatabaseName);
+    const dispatch = useAppDispatch();
+    const isSelected = useAppSelector(documentSchemaSelectors.isSelectedCollectionName(collectionName));
+    const validatorsAll = useAppSelector(documentSchemaSelectors.allValidators);
+    const { databasesService } = useServices();
+    const { value: isDeleteModalOpen, toggle: toggleDeleteModal } = useBoolean(false);
+
+    const asyncSaveValidators = useAsyncCallback(async (items: DocumentSchemaValidatorConfig[]) => {
+        await databasesService.saveSchemaValidation(
+            databaseName,
+            documentSchemaUtils.mapToSchemaValidationConfigurationDto(items)
+        );
+        dispatch(documentSchemaActions.validatorsSaved());
+    });
+
+    const form = useForm<DocumentSchemaFormData>({
+        resolver: yupResolver(formSchema),
+        defaultValues: {
+            collection: collectionName,
+            schema: jsonUtil.formatJson(schema),
+        },
+    });
+
+    const { handleSubmit, reset } = form;
+
+    const handleEditSchema = async (data: DocumentSchemaFormData) => {
+        const updatedItem = documentSchemaUtils.maptoDocumentSchemaValidatorConfigDto(data);
+        dispatch(documentSchemaActions.validatorEdited(updatedItem));
+        const next = [...validatorsAll.filter((v) => v.Name !== updatedItem.Name), updatedItem];
+        await asyncSaveValidators.execute(next);
```

Comment:
You dont need `asyncSaveValidators` here. Wrap it in `tryHandleSubmit` and use `form.formState.isSubmitting` for loading state. It will also automaticly disable all inputs.